presto-main/src/main/java/com/facebook/presto/split/BufferingSplitSource.java

```diff
@@ -45,6 +45,12 @@ public ConnectorId getConnectorId()
     }
 
     @Override
+    public Split getEmptySplit()
```

Comment:
Would it be that simple? Adding a connector, connector factory, dragging a creation of that through whole system? I'm not familiar with that part of the code, but it looks like `InformationSchema` and `system` connectors are treated specially in dozen of places (like `ConnectorManager`, `Catalog`, `TransactionManager`, ...), although I don't understand why they are so special.

Could you take a look at current implementation version? It depends on special handling of `EmptySplit` (extra 3 lines in `Driver`), but still seems simpler to me, compared to adding special connector for that.